src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -133,6 +134,7 @@
 
   private static final Pattern reSimpleName = Pattern.compile("^[a-zA-Z_][a-zA-Z_0-9]*$");
   private ThreadPoolExecutor searchThreadPoolExecutor;
+  private Keyable keyableField;
```

Comment:
It might be better track this as a FieldDef, and verify it is instanceof Keyable when set. Then you will have access to the field name and any other properties.

Reply:
A good reason to have this as Keyable is that I could directly use it to for `getTerm()`, instead of having to cast it again while updating the documents. Name is the only field that is needed now, but since we made the interface have the name which is anyway defined for all FieldDefs, we would get access to it directly.

If it's okay, we could change this to be a `FieldDef` later if we need access to other properties.

Reply:
The idea was to allow access to the name without creating a duplicate in the interface. This is fine for now.

I'm thinking this could maybe be done better by giving field properties a common interface to get their FieldDef, but that is future work.